R-package/src/install.libs.R

```diff
@@ -99,6 +99,32 @@ if (!use_precompile) {
 
   # Install
   system(paste0(cmake_cmd, " .."))
+
+  # R CMD check complains about the .NOTPARALLEL directive created in the cmake
+  # Makefile. We don't need it here anyway since targets are built serially, so trying
+  # to remove it with this hack
```

Comment:
Is it safe? I remember setting `-j4` allowed to speed up building significantly. Forgive me if this hack is not about that.
https://github.com/microsoft/LightGBM/blob/6129208006c91f9ac94891690aca6f0c49d9b08e/.ci/test.sh#L151

Reply:
I don't know why the `Makefile` being generated sometimes has the `.NOTPARALLEL` directive in it, but I know that that is non-portable which causes `R CMD CHECK` to complain. Removing this might make using parallelization with `-j4` unsafe, but I think it is ok. Right now with the `Rscript build_r.R` process, we are not giving users any ability to customize the command e.g. to choose `make _lightgbm` vs. `make _lightgbm -j4`. I think it's ok to hack around this to get us closer to CRAN, and come back to it if / when we introduce a build process for R that uses parallelization when creating `liblightgbm`.

Reply:
Agree!